pkg/csource/options.go

```diff
@@ -204,63 +204,55 @@ func (opts Options) Serialize() []byte {
 	return data
 }
 
-func deserializeLegacyOptions1(data string, opts *Options) error {
-	waitRepeat, debug := false, false
-	n, err := fmt.Sscanf(data,
-		"{Threaded:%t Collide:%t Repeat:%t Procs:%d Sandbox:%s"+
-			" Fault:%t FaultCall:%d FaultNth:%d EnableTun:%t UseTmpDir:%t"+
-			" HandleSegv:%t WaitRepeat:%t Debug:%t Repro:%t}",
-		&opts.Threaded, &opts.Collide, &opts.Repeat, &opts.Procs, &opts.Sandbox,
-		&opts.Fault, &opts.FaultCall, &opts.FaultNth, &opts.NetInjection, &opts.UseTmpDir,
-		&opts.HandleSegv, &waitRepeat, &debug, &opts.Repro)
-
-	if err == nil {
-		if want := 14; n != want {
-			return fmt.Errorf("failed to parse repro options: got %v fields, want %v", n, want)
-		}
+func deserializeLegacyOptions(data string, opts *Options) (int, error) {
+	ignoreBool := true
+	keyToTarget := map[string]struct {
+		typeFormat  string
```

Comment:
I wonder if it might work without specifying format for every individual type. In the end, we only use default formats for every type, so `Sscanf` should be able to guess it via reflection.

Reply:
Nice catch. It works.